RFC_MULTI_DATABASE_SUPPORT.md

```diff
@@ -0,0 +1,87 @@
+# RFC: Multi-Database Support for Nandi
+
+**Status:** Draft | **Issue:** [#137](https://github.com/gocardless/nandi/issues/137) | **Date:** 2025-08-07
+
+## Summary
+
+Adds multi-database support to Nandi with database declaration syntax. Zero breaking changes - existing single-database projects work unchanged.
+
+## API Design
+
+### Database Declaration in Migrations
+```ruby
+class CreateUsersTable < Nandi::Migration
+  database :primary  # Target specific database
+  
+  def up
+    create_table :users do |t|
+      t.string :email
+    end
+  end
+end
+```
+
+### Multi-Database Configuration
+```ruby
+Nandi.configure do |config|
+  config.databases = {
+    primary: { migration_directory: "db/safe_migrations/primary" },
+    analytics: { 
+      migration_directory: "db/safe_migrations/analytics",
+      output_directory: "db/migrate/analytics" 
+    }
+  }
+end
+```
+
+### Generator Support
+```bash
+# Generate database-specific migration
+rails generate nandi:migration create_reports --database=analytics
+
+# Compile all databases
+rails generate nandi:compile
+```
+
+## Implementation
+
+### Core Changes
+- **Nandi::Migration**: Added `database(name)` class method and `target_database` accessor
+- **Nandi::Config**: Added `databases` hash with helper methods (`migration_directory_for`, `output_directory_for`, `multi_database?`)
+- **Nandi::CompiledMigration**: Routes migrations to database-specific output directories
+- **Generators**: Enhanced with `--database` option for targeting specific databases
+
+### Enhanced Lockfile (.nandilock.yml)
+```yaml
+# Database-specific entries (new)
+primary/20240101120000_create_users.rb:
```

Comment:
Should we have`.name?_nandilock.yml` files? Similar to above, omit the name for the default database, and existing single db configs are not affected. And then inside, we don't specify the folder name to keep things consistent across both scenarios